app/src/main/java/de/tum/in/tumcampusapp/component/ui/cafeteria/details/OpenHoursHelper.java

```diff
@@ -44,8 +44,15 @@ public String getHoursByIdAsString(int id, DateTime date) {
         int dayOfWeek = date.getDayOfWeek();
 
         //Split up the data string from the database with regex which has the format: "Mo-Do 11-14, Fr 11-13.45" or "Mo-Fr 9-20"
-        Matcher m = Pattern.compile("([a-z]{2}?)[-]?([a-z]{2}?)? ([0-9]{1,2}(?:[\\.][0-9]{2}?)?)-([0-9]{1,2}(?:[\\.][0-9]{2}?)?)", Pattern.CASE_INSENSITIVE)
-                .matcher(result);
+        Matcher m;
+        if (context.getString(R.string.language).equals("de")){
+            m = Pattern.compile("([a-z]{2}?)[-]?([a-z]{2}?)? ([0-9]{1,2}(?:[\\:][0-9]{2}?)?)-([0-9]{1,2}(?:[\\:][0-9]{2}?)?)", Pattern.CASE_INSENSITIVE)
+                       .matcher(result);
+        } else {
+            // use three letter shortenings for english: Mon, Tue, Wed, Thu, Fri, Sat, Sun
+            m = Pattern.compile("([a-z]{3}?)[-]?([a-z]{3}?)? ([0-9]{1,2}(?:[\\:][0-9]{2}?)?)-([0-9]{1,2}(?:[\\:][0-9]{2}?)?)", Pattern.CASE_INSENSITIVE)
+                               .matcher(result);
+        }
```

Comment:
Sad that we were not able to get rid of this yet. This is really one of the most awful places in the whole code base :crying_cat_face: